### DISCUSSION TITLE
Consider defaulting a var/let declaration to auto type

### DISCUSSION THREAD

**Discussion by giucam** [carbon-language/carbon-lang#1495] [Language design]
In order to make a var/let declaration with type inference one needs to specify the type as `auto`, as in:
```
var foo: auto = 3;
```

It seems to me it would be more natural if you could just do
```
var foo = 3;
```

I think the `auto` there does not really do anything useful,  while just being one more thing to type.

**Comment by sanxiyn** [comment]
FYI, it was considered, see [proposal p0851](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0851.md), especially "Alternatives considered" > "Elide the type instead of using auto".

**Reply by GiorgosAlexakis** [reply]
It looks like the proposal was accepted??

**Reply by giucam** [reply]
The proposal lists these as disadvantages of eliding the type:

> - Type inference becomes the lack of a type, rather than the presence of something different.
> - C++'s syntax legacy may have needed auto in order to provide type inference, where Carbon does not.

I don't see why they would be disadvantages, the second one even seems an advantage to me, unless i misunderstand it.

> - Removes syntactic distinction between binding of a new name and reference to an existing name, [particularly for pattern matching](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0851.md#pattern-matching). Reintroducing this distinction would likely require additional syntax, such as Swift's nested let.

I can see though that this one is a valid concern, although having to specify `var` in a value binding pattern doesn't seem like a problem to me.

**Reply by jonmeow** [reply]
> I don't see why they would be disadvantages, the second one even seems an advantage to me, unless i misunderstand it.

Oops, that looks like my mistake.

But I think the leads were really focused on type inference becoming the _lack_ of a name, and generally wanting to encourage writing types explicitly. I don't know how much that'll shift as more feedback arrives.
